File: lib_fly_db/src/main/java/com/fly/tour/db/entity/NewsType.java
package com.fly.tour.db.entity;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Description: <NewsType><br>
 * Author:      mxdl<br>
 * Date:        2019/5/27<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
public class NewsType implements Parcelable {
    public int id;
    public String typename;
    public String addtime;
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTypename() {
        return typename;
    }

    public void setTypename(String typename) {
        this.typename = typename;
    }

    public String getAddtime() {
        return addtime;
    }

    public void setAddtime(String addtime) {
        this.addtime = addtime;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.id);
        dest.writeString(this.typename);
        dest.writeString(this.addtime);
    }

    public NewsType() {
    }

    protected NewsType(Parcel in) {
        this.id = in.readInt();
        this.typename = in.readString();
        this.addtime = in.readString();
    }

    public static final Creator<NewsType> CREATOR = new Creator<NewsType>() {
        @Override
        public NewsType createFromParcel(Parcel source) {
            return new NewsType(source);
        }

        @Override
        public NewsType[] newArray(int size) {
            return new NewsType[size];
        }
    };

    @Override
    public String toString() {
        return "NewsType{" + "id=" + id + ", typename='" + typename + '\'' + ", addtime='" + addtime + '\'' + '}';
    }
}

File: lib_fly_db/src/main/java/com/fly/tour/db/entity/NewsType.kt
package com.fly.tour.db.entity

import android.annotation.SuppressLint
import android.os.Parcel
import android.os.Parcelable

/**
 * Description: <NewsType><br>
 * Author:      mxdl<br>
 * Date:        2019/5/27<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
class NewsType : Parcelable {
    var id: Int = 0
    var typename: String? = null
    var addtime: String? = null

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(this.id)
        dest.writeString(this.typename)
        dest.writeString(this.addtime)
    }

    constructor() {}

    protected constructor(`in`: Parcel) {
        this.id = `in`.readInt()
        this.typename = `in`.readString()
        this.addtime = `in`.readString()
    }

    override fun toString(): String {
        return "NewsType{id=$id, typename='$typename', addtime='$addtime'}"
    }

    companion object {

        @SuppressLint("ParcelCreator")
        val CREATOR: Parcelable.Creator<NewsType> = object : Parcelable.Creator<NewsType> {
            override fun createFromParcel(source: Parcel): NewsType {
                return NewsType(source)
            }

            override fun newArray(size: Int): Array<NewsType?> {
                return arrayOfNulls(size)
            }
        }
    }
}
